### PR TITLE
ENH - reorganised sphere_mesh and renamed into mesh_spere. removed ks…

### PR DESCRIPTION
**Pull request by schoffelen**
…phere and msphere as separate functions

### CODE REVIEW

**Comment by schoffelen** [comment]
And the plot thickens...

As discussed with @robertoostenveld sphere_mesh has been renamed into mesh_sphere, and the function has been rewritten somewhat, for a more logical flow. msphere and ksphere have been removed as separate functions and are now subfunction to mesh_sphere. 

Mesh_sphere needs to be completed with the 'tetrahedron', and 'octahedron' methods, respectively.

@StolkArjen we also discussed the cylinder function. For consistency, we'd want this to be called mesh_cylinder, to avoid conflicts with the MATLAB built-in version (that does not produce a mesh). I will add this as a commit into this PR.

I also removed all occurrences of icosahedron as separate function, and updated synchronize-private.

**Comment by robertoostenveld** [comment]
looks fine to me. Please move ahead

**Comment by schoffelen** [comment]
tja, zoiets dus?

File: external/bemcp/mesh_icosahedron.m [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+pos(2:6, 2) = rho*sin(phi)';
+pos(2:6, 3) = rho/2;
+
+pos(7:11, 1) = rho*cos(phi - pi/5)';
+pos(7:11, 2) = rho*sin(phi - pi/5)';
+pos(7:11, 3) = -rho/2;
+
+pos(12, :) = [0 0 -1];          % bottom point
+
+if n>0
```
Comment (robertoostenveld) [inline external/bemcp/mesh_icosahedron.m]: I would remove the refinement here.  That is meant to make more spherical meshes, and hence done in mesh_sphere.

File: external/bemcp/mesh_icosahedron.m [hunk 2]
```diff
@@ -0,0 +1,84 @@
+function [pos, tri] = mesh_icosahedron(n)
```
Comment (robertoostenveld) [inline external/bemcp/mesh_icosahedron.m]: no input argument needed

File: external/dipoli/mesh_icosahedron.m [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+pos(7:11, 3) = -rho/2;
+
+pos(12, :) = [0 0 -1];          % bottom point
+
+if n>0
+  % perform an n-fold refinement
+  for i=1:n
+    [pos, tri] = refine(pos, tri);
+    % scale all vertices to the unit sphere
+    pos = pos ./ repmat(sqrt(sum(pos.^2,2)), 1, 3);
```
Comment (robertoostenveld) [inline external/dipoli/mesh_icosahedron.m]: did you check the unit-scaling in case no refinement is done?

**Comment by robertoostenveld** [comment]
> tja, zoiets dus?

Ik heb alleen nog kleine suggestie, zie comment.

File: external/dipoli/mesh_icosahedron.m (hunk 3 above)
Comment (schoffelen) [reply external/dipoli/mesh_icosahedron.m]: Now I have :)
It's OK as is

**Pull request merged**